Grammar/python.gram

```diff
@@ -1081,11 +1081,16 @@ invalid_legacy_expression:
             "Missing parentheses in call to '%U'. Did you mean %U(...)?", a->v.Name.id, a->v.Name.id) : NULL}
 
 invalid_expression:
+    # Soft keywords need to be ignored in these errors because they can be parsed as NAME NAME
     # !(NAME STRING) is not matched so we don't show this error with some invalid string prefixes like: kf"dsfsdf"
-    # Soft keywords need to also be ignored because they can be parsed as NAME NAME
    | !(NAME STRING | SOFT_KEYWORD) a=disjunction b=expression_without_invalid {
         _PyPegen_check_legacy_stmt(p, a) ? NULL : p->tokens[p->mark-1]->level == 0 ? NULL :
         RAISE_SYNTAX_ERROR_KNOWN_RANGE(a, b, "invalid syntax. Perhaps you forgot a comma?") }
+    # Literals are also ignored in this error because an attribute access to a literal would be very much useless
```

Comment:
We have changed these kinds of reports due to an increased number of false positives, as they tend to happen when checking for subexpressions in other grammar parts. 

Additionally, the problem with the string prefixes is that they are identified by the tokenizer, so adding these in the parser can be quite problematic to get correct and synchronize, because the tokenizer can interrupt the prefix in the second pass earlier

Reply:
One example why the invalid prefix will fail if made on the parser: the prefix must be together with the string. For example, in 3.10 if you add the prefix with a space in the middle, you get:

```
f>>> f "ijooi"
  File "<stdin>", line 1
    f "ijooi"
      ^^^^^^^
SyntaxError: invalid syntax
```

but with this patch you will get:

```
>>> f "ijooi"
  File "<stdin>", line 1
    f "ijooi"
      ^^^^^^^
SyntaxError: invalid syntax
```

This will cause bad identifications for example:

```
>>> ["oijoijio", "joijoi", my_variable "iojioioj"]
  File "<stdin>", line 1
    ["oijoijio", "joijoi", my_variable "iojioioj"]
                           ^^^^^^^^^^^
SyntaxError: invalid string prefix 'my_variable'
```

There the problem is not that `my_variable` was attempted as a prefix, but that the user forgot a comma.

Reply:
> There the problem is not that my_variable was attempted as a prefix, but that the user forgot a comma.

Implemented this in the tokenizer:
```py
>>> ["oijoijio", "joijoi", my_variable "iojioioj"]
  File "<stdin>", line 1
    ["oijoijio", "joijoi", my_variable "iojioioj"]
                                       ^^^^^^^^^^
SyntaxError: invalid syntax
>>> ["oijoijio", "joijoi", my_variable"iojioioj"]
  File "<stdin>", line 1
    ["oijoijio", "joijoi", my_variable"iojioioj"]
                           ^^^^^^^^^^^
SyntaxError: invalid string prefix 'my_variable'. Perhaps you forgot a comma?
```